source/common/router/router.cc

```diff
@@ -113,6 +113,11 @@ bool convertRequestHeadersForInternalRedirect(Http::RequestHeaderMap& downstream
 
 constexpr uint64_t TimeoutPrecisionFactor = 100;
 
+bool shouldProxy(Http::RequestHeaderMap& headers, const RouteEntry* route_entry) {
```

Comment:
fair - I'd been trying to avoid TcpProxying places since some day hopefully we'll be able to tunnel many "TCP" payloads over QUIC CONNECT (which is awesome in all sorts of ways) but I think avoiding it in API and include is good enough - we can easily change helper names in the router class :-)